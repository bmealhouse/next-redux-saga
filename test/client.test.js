import React from 'react'
import withRedux from 'next-redux-wrapper'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import withReduxSaga from '..'
import ClassComponent from './components/class-component'
import FunctionalComponent from './components/functional-component'
import AsyncGetInitialProps from './components/async-get-initial-props'
import SyncGetInitialProps from './components/sync-get-initial-props'
import configureStore from './store/configure-store'
import createSnapshot from './utils/create-snapshot'
import getInitialProps from './utils/get-initial-props'
import {ASYNC_REDUX_SAGA_PROP_TEXT,STATIC_PROP_TEXT, SYNC_REDUX_PROP_TEXT} from './constants'

test('Wrapped component passes along React props', () => {
  const WrappedComponent = withRedux(configureStore)(
    withReduxSaga()(FunctionalComponent),
  )

  const wrapper = shallow(<WrappedComponent/>).dive()
  expect(toJson(wrapper)).toMatchSnapshot()
})

test('Wrapped component skips getInitialProps when it does not exist', async () => {
  const WrappedComponent = withRedux(configureStore)(
    withReduxSaga()(ClassComponent),
  )

  const props = await getInitialProps(WrappedComponent)
  expect(props.isServer).toBeFalsy()

  createSnapshot(WrappedComponent, props)
})

test('Wrapped component awaits synchronous getInitialProps', async () => {
  const WrappedComponent = withRedux(configureStore)(
    withReduxSaga()(SyncGetInitialProps),
  )

  const props = await getInitialProps(WrappedComponent)

  expect(props.isServer).toBeFalsy()
  expect(props.initialState).toEqual({syncReduxProp: SYNC_REDUX_PROP_TEXT})
  expect(props.initialProps).toEqual({staticProp: STATIC_PROP_TEXT})

  createSnapshot(WrappedComponent, props)
})

test('Wrapped component awaits asynchronous getInitialProps', async () => {
  const WrappedComponent = withRedux(configureStore)(
    withReduxSaga()(AsyncGetInitialProps),
  )

  const props = await getInitialProps(WrappedComponent)

  expect(props.isServer).toBeFalsy()
  expect(props.initialState).toEqual({syncReduxProp: SYNC_REDUX_PROP_TEXT})
  expect(props.initialProps).toEqual({staticProp: STATIC_PROP_TEXT})

  createSnapshot(WrappedComponent, props)
})
