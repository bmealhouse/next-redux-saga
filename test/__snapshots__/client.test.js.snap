// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapped component awaits asynchronous getInitialProps (async) 1`] = `
<div>
  AsyncGetInitialProps({"mode":"async","staticProp":"Static message from getInitialProps()","syncReduxProp":"Synchronous message from Redux"})
</div>
`;

exports[`Wrapped component awaits asynchronous getInitialProps (sync) 1`] = `
<div>
  AsyncGetInitialProps({"mode":"sync","staticProp":"Static message from getInitialProps()","syncReduxProp":"Synchronous message from Redux","asyncReduxSagaProp":"Asynchronous message from Redux-Saga"})
</div>
`;

exports[`Wrapped component awaits synchronous getInitialProps (async) 1`] = `
<div>
  SyncGetInitialProps({"mode":"async","staticProp":"Static message from getInitialProps()","syncReduxProp":"Synchronous message from Redux"})
</div>
`;

exports[`Wrapped component awaits synchronous getInitialProps (sync) 1`] = `
<div>
  SyncGetInitialProps({"mode":"sync","staticProp":"Static message from getInitialProps()","syncReduxProp":"Synchronous message from Redux"})
</div>
`;

exports[`Wrapped component passes along React props (async) 1`] = `
<div>
  FunctionalComponent({"mode":"async"})
</div>
`;

exports[`Wrapped component passes along React props (sync) 1`] = `
<div>
  FunctionalComponent({"mode":"sync"})
</div>
`;

exports[`Wrapped component skips getInitialProps when it does not exist (async) 1`] = `
<div>
  ClassComponent({"mode":"async","store":{"sagaTask":{"@@redux-saga/TASK":true,"id":4,"name":"rootSaga","joiners":[],"done":{}}},"isServer":false})
</div>
`;

exports[`Wrapped component skips getInitialProps when it does not exist (sync) 1`] = `
<div>
  ClassComponent({"mode":"sync","store":{"sagaTask":{"@@redux-saga/TASK":true,"id":4,"name":"rootSaga","joiners":[],"done":{}}},"isServer":false})
</div>
`;
